### ISSUE TITLE
Can't install on Samsung Galaxy i9100 with CyanogenMod 12.1

### ISSUE THREAD

**Issue by karstengit** [guardianproject/lildebi#197]
I think you need the log created.
Here it is:

manufacturer: samsung
model: GT-I9100
product: GT-I9100
brand: Samsung
device: GT-I9100
board: smdk4210
ID: LMY48Y
CPU ABI: armeabi-v7a
release: 5.1.1
incremental: 94e1372c6d
codename: REL
SDK: 22

install.log:

# cd /data/data/info.guardianproject.lildebi/app_bin
# ./create-debian-setup.sh  /data/data/info.guardianproject.lildebi/app_bin /storage/sdcard0 /storage/sdcard0/debian.img /data/debian \
&& ./stop-debian.sh  /data/data/info.guardianproject.lildebi/app_bin /storage/sdcard0 /storage/sdcard0/debian.img /data/debian \
&& ./unmounted-install-tweaks.sh  /data/data/info.guardianproject.lildebi/app_bin /storage/sdcard0 /storage/sdcard0/debian.img /data/debian \
&& ./start-debian.sh  /data/data/info.guardianproject.lildebi/app_bin /storage/sdcard0 /storage/sdcard0/debian.img /data/debian \
&& /data/data/info.guardianproject.lildebi/app_bin/chroot /data/debian /data/data/info.guardianproject.lildebi/app_bin/complete-debian-setup.sh  /data/data/info.guardianproject.lildebi/app_bin /storage/sdcard0 /storage/sdcard0/debian.img /data/debian 
# ./stop-debian.sh  /data/data/info.guardianproject.lildebi/app_bin /storage/sdcard0 /storage/sdcard0/debian.img /data/debian 
# exit
Cannot execute /data/data/info.guardianproject.lildebi/app_bin/sh: No such file or directory

**Comment by karstengit** [comment]
[install.log.zip](https://github.com/guardianproject/lildebi/files/92229/install.log.zip)

**Comment by leokitten** [comment]
I have the very same problem on a:

- Samsung Galaxy S3 Mini / GT-i8190
- CyanogenMod version: 12.1-20160209-UNOFFICIAL-golden (From: http://maclaw.pl )
- Android version: 5.1.1
- The device is rooted. Verified by the "Root Verifier" app.
- Google Store/Play is not installed

**Comment by karstengit** [comment]
I could already analyze the error that occurs with the debian kit of Sven Ola - but i could not fix it.
http://www.apk20.com/apk/231695/ (original http://sven-ola.dyndns.org/repo/debian-kit-en.html)

Here was the problem that after Cyanogenmod 10 the busybox does not correct support the busybox ash.
https://en.wikipedia.org/wiki/Almquist_shell
http://lists.busybox.net/pipermail/busybox/2012-June/078044.html

Maybe here it is an similar problem?
Does this project use ash shell?

leokitten, have you found another Debian implementation that works with Cyanogenmod?
I could find here another variation i will try: http://forums.debian.net/viewtopic.php?f=16&t=106246

**Comment by leokitten** [comment]
I'm not a Linux expert, but I guess that the problem is that it searches for system files in this folder:
/data/data/info.guardianproject.lildebi/app_bin/

when I linked the "sh" there, with a symbolic link, it went further and asked for more system commands to be found there. If this is a shell script, then maybe just the $PATH is not set up correctly.

**Comment by leokitten** [comment]
(i'll check it tomorrow night)

**Comment by karstengit** [comment]
Yes - the problem is that this Debian implementations are not easy to understand.
It must be embedded into the Android system using it's kernel.
This is done with scripting that is not documented (solution Sven-Ola), but it is using tricky programming. :-(
But i have same limited shell knowledge that maybe will help.

I did not have a closer look on the lildebi solution, but maybe we can find the problem together?
Can you say me anything about the installation process?

It's a pity that there is no support by the authors of this solutions ...

**Comment by leokitten** [comment]
i gave this up, and since i had other problems too (like the GPS isn't working) i've ordered an Ubuntu Touch phone (BQ Aquaris E5 HD Ubuntu Edition). I'll be curious how it'll work out.

**Comment by karstengit** [comment]
Don't give up! :-)
The i9100 is working good with CM-11-20141115.
One of the little bugs is that you can't activate the GPS with the short menu.
You must go into the settings and switch it on. Then it is working good.

This Ubuntu phone sounds interesting - but you must spent a lot of money for it.
Do you have an X11 and an native linux distro direct that is usable?

**Comment by leokitten** [comment]
Now i'm reinstalling CM with a total wipe/reformat. I read somewhere that this can make a difference. And also installing the google apps (from the same source).

My desktop is windows, but the docking station must also work if I run an Ubuntu in a virtualbox, because that's just USB.

**Comment by leokitten** [comment]
cool now the GPS works. Will try to install Busybox before lil debi.

**Comment by leokitten** [comment]
- The phone was previously rooted using this guide: http://novafusion.pl/tutorials/installation-guide-golden/
- Booted into recovery mode, wiped everything and reformatted it
- Installed CM 12.1: http://novafusion.pl/tutorials/installation-guide-golden/
- Installed Google Apps (the "full" package): http://novafusion.pl/tutorials/how-to-install-google-apps/
- After reboot and basic configuration: Waited until google updates itself (took some time)
- Enabled the developer tools. Enabled root access. Enabled the local terminal.
- Installed Busybox (free) from the Google Play Store
- Tried to install Lil' Debi: same error

But I saw that the Google Store had some other Debian solutions. Although those are not chroot I guess.

**Comment by karstengit** [comment]
CM12 is crashing the camera. https://jira.cyanogenmod.org/browse/CYAN-7358
Do you have the same problem?
I did not install the Google stuff because i used Cyanogenmod to have a mobile that will keep my privacy. ;-)

Besides - what is the GSM / 3G base of the Ubuntu phone - is it Android?

**Comment by leokitten** [comment]
Unfortunately it seems to me that Android based systems (CM also) are only stable when the Google apps are installed, and when instead of the "Trebuchet" launcher one selects the Google launcher. For example my other problem was that Skype couldn't use the microphone, but when I switched to the Google launcher, it just worked! And a lot of apps worked only when they were installed from the google store.

I read a developer comment about "Ubuntu Touch" that said, in the beginning when they started the project it was CM/Android based, but later they have made Ubuntu the underlying system instead of Android. So in theory it's running natively, and not in a virtual environment of an Android framework.

**Comment by leokitten** [comment]
(BTW the camera works fine for me. I've opened and closed it multiple times.)

**Comment by karstengit** [comment]
My system crashed after taking a photo.
Special after opening, closing, reopening and then taking photos after some usage.

I am using CM11 without the Google package and i have installed F-Droid instead.
I have no problems. When i really want to install an package from the Google store i download it over a website, push it to the SD and install it. This works.

But of course the sense of the Google world is to stay in it "to be safe that everything works fine" - including grabbing all your personal data. ;-)
So my decision was to keep privacy and instead live with the problems that parts are not working "perfect".

When the Ubuntu phone is without Android this would be an argument for me to buy it!
But i could not find any clear facts about this.

Another problem seems to be the sources of the GSM and UMTS stuff for interacting with the radio network.
Normally you only get it compiled, and this parts are used in compiled form from the manufacturers in Cyanogenmod.
Getting the source would enable to use the phone for every purpose you want, including a walkie talkie or hacking the network. ;-)

**Comment by karstengit** [comment]
**Another idea for experiments:**
https://wiki.ubuntu.com/Touch/Devices
https://wiki.ubuntu.com/Touch/Devices/i9100 but many things are not working
( Cellular Radio 3G is working, but calls/SMS are not )

**Source code for the Ubuntu phone:**
https://github.com/bq/aquaris-E5/tree/aquaris-E5/kernel/drivers
I will try to find the drivers in it ...

http://www.gizmag.com/ubuntu-edge-smartphone-computer/28481/
Hmm - is this the same for the "BQ Aquaris E5 HD" ?

**Comment by godlessfather** [comment]
IRC may be a better place for the last several comments. Lildebi's latest
code has been broken due to busybox and gpgv compilation issues.

**Comment by karstengit** [comment]
Thanks for the tip my godlessfather!

Where can i find details about this problems with busybox?
Maybe this is the general reason for all the Debian implementations?

**Comment by godlessfather** [comment]
Android 5+ requires binaries to be compiled differently than previous
releases. The reason things break is that our current way of building
busybox addresses that requirement but doesn't work with Android's way of
networking. Simply putting CM's busybox in place of lildebi's fixes that.
You're still left with gpgv failure. I'm not sure what's going on with it.
Check out other issues and the comments such as
https://github.com/guardianproject/lildebi/issues/196#issuecomment-170257866.


Anyone better than me at coding, please feel free to investigate PIE/static
compilation of busybox and gpgv to solve this persistent issue. There's a
reason some busybox binaries work with network commands like wget bit this
one doesn't. Is it bionic libc vs. glibc? Why does gpgv fail? Issue #176
and #159 mention this.

**Comment by Rbagman** [comment]
Same problem.

I have: Rooted; Samsung Galaxy Note 2 (GT-N7100/Android 4.4.2); SuperSU Pro.
